aci/resource_rest.go

```diff
@@ -43,6 +47,7 @@ func resourceAciRest() *schema.Resource {
 			"dn": &schema.Schema{
 				Type:     schema.TypeString,
 				Computed: true,
+				ForceNew: true,
```

Comment:
Dn is something that should be computed only, we don't want users to pass the Dn. It's never used in the POST/PUT calls. Please remove all the references which checks for the diff in dn and also remove the ForceNew: true from the dn param as well.

Reply:
This force new is required for the resource to be recreated when the resourceAciRestCustomizeDiff function tags the dn as changed after apply. I will remove the changes concerning the dn attribute.  However,  this pull request will no longer completely fix the issue as changes to the dn attribute will not recreate the corresponding managed object. I will ping you back when it's done.

Reply:
The DN of an object is always based on one of the fields of the object so if the content is different we should be able to infer that the object need to be recreated. Am I missing something?

Reply:
The DN is computed only after the apply, so when Terraform is made aware of the change, it is already to late to recreate the object.

The resourceAciRestCustomizeDiff function computes if the apply will change the dn and mark it as such during the plan phase of the terraform lifecycle.

But if this patch is too involved to review, we (my org) can live a patch that only recreate the object when the path change as the dn can be included in it most situation that we care about.

Reply:
My point here is, Dn is not user-configurable, if you pass the dn to this resource in the terraform file we are not going to pass it to the API, the only thing passed to api is content/payload and the path. Dn we calculate based on the object we are going to create. While using Payload mode to create the objects using rest, dn won't be set at all, because we will never know what user tried to do there. There is no guarantee that user will pass the dn in content/payload. So checking the changes in dn is not feasible in this case. If there is something needs to be re-created we might need to think any other way as this resource was never designed to be 100% idempotent based on the nature of content it's going to handle. 


The very simple thing here is you don't need Dn to create an object in the ACI, you need the classname and required attributes, Dn is calculated automatically, So if you change any attribute which requires an object to be recreated checking the changes in Dn won't help as it might not be set in some cases.

Reply:
I did the revert and removed the force new on the dn attribute. Thanks for your explanation and sorrry about the delay, we were busy migrating our data-center networks in aci.